### DISCUSSION TITLE
Typing extracted states

### DISCUSSION THREAD

**Discussion by BrunoQuaresma** [statelyai/xstate#3665] [Q&A] [answered]
Hi folks! I have multiple machines that load some API data and I would like to extract some states to reuse them but I find it very hard to create types for it. Is there a way to type a "state" outside of a machine?

**Comment by Andarist** [comment, marked as answer]
At the moment this isn't quite possible if you want to use the Studio for bidirectional editing and the typegen for stricter types. Reusing states like this poses some additional UX concerns that we need to think through before implementing support for this.
